wpilibj/src/shared/java/edu/wpi/first/wpilibj/CircularBuffer.java

```diff
@@ -94,15 +96,70 @@ public double popBack() {
     }
 
     m_length--;
-    return m_data[(m_front + m_length) % m_data.length];
+    return m_data.get((m_front + m_length) % m_data.size());
+  }
+
+  /**
+   * Resizes internal buffer to given size.
+   */
+  void resize(int size) {
+    if (size > m_data.size()) {
```

Comment:
It's not too bad to rewrite this to just new a fresh double[] and copy the existing elements into it into the right places.  We rarely resize, so this shouldn't be a big deal.